src/app/thumbnails.h

```diff
@@ -1,56 +1,23 @@
-// Aseprite
-// Copyright (C) 2016  Carlo Caputo
-//
-// This program is free software; you can redistribute it and/or modify
-// it under the terms of the GNU General Public License version 2 as
-// published by the Free Software Foundation.
-
 #ifndef APP_THUMBNAILS_H_INCLUDED
 #define APP_THUMBNAILS_H_INCLUDED
 #pragma once
 
-#include "base/shared_ptr.h"
-#include "base/connection.h"
-#include "she/surface.h"
-#include "doc/object_id.h"
-#include "doc/object.h"
-#include "doc/frame.h"
-
-#include <map>
-#include <set>
-#include <utility>
+#include "gfx/rect.h"
 
 namespace doc {
   class Cel;
-  class Image;
-  class Document;
 }
 
-namespace app {
-  class Document;
+namespace she {
+  class Surface;
+}
 
+namespace app {
   namespace thumb {
 
-    typedef std::pair< int, int > Dimension;
-
-    class Request {
-    public:
-      Request();
-      Request(const app::Document* doc, const doc::frame_t frm, const doc::Image* img, const Dimension dim);
-      const app::Document* document;
-      const doc::frame_t frame;
-      const doc::Image* image;
-      const Dimension dimension;
-    };
-
-    class SurfaceData {
-    public:
-      static she::Surface* generate(Request& req);
-      static she::Surface* fetch(const app::Document* doc, const doc::Cel* cel, const gfx::Rect& bounds);
-    };
-
-  } // thumbnails
+    she::Surface* get_surface(const doc::Cel* cel, const gfx::Rect& bounds);
```

Comment:
Definitely I'd like this function to be called `app::thumb::get_cel_thumbnail` (easier to `grep`). In a future I'll move the file system thumbnails into the `app::thumb` namespace and we can have something like `app::thumb::get_file_thumbnail`.